sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResult.java

```diff
@@ -247,4 +261,37 @@ private void writeNonNullObjectField(JsonGenerator jsonGenerator, String fieldNa
              jsonGenerator.writeObjectField(fieldName, object);
         }
     }
+
+    /**
+     * Private copy constructor, only to be used internally for serialization purposes
+     * <p>
+     * NOTE: This constructor does not copy all the fields to avoid memory issues.
+     */
+    private StoreResult(StoreResult storeResult) {
+        this.storeResponse = StoreResponse.createSerializableStoreResponse(storeResult.storeResponse);
+        this.exception = ImplementationBridgeHelpers
+            .CosmosExceptionHelper
+            .getCosmosExceptionAccessor()
+            .createSerializableCosmosException(storeResult.exception);
+        this.lsn = storeResult.lsn;
+        this.partitionKeyRangeId = storeResult.partitionKeyRangeId;
+        this.quorumAckedLSN = storeResult.quorumAckedLSN;
+        this.globalCommittedLSN = storeResult.globalCommittedLSN;
+        this.numberOfReadRegions = storeResult.numberOfReadRegions;
+        this.itemLSN = storeResult.itemLSN;
+        this.sessionToken = storeResult.sessionToken;
```

Comment:
Create a new object or convert to string may be?

Reply:
This is a good point Milis. I think it is worth doing in this iteration, as SessionToken implementation contains some maps, which will consume high memory. Good catch!

Reply:
Fixed it, please check!